### PR TITLE
Switching MathInternal.Min/Max to managed implementations

### PR DESCRIPTION
**Pull request by CoryCharlton**
## Description
- Switching min/max to managed implementations in MathInternal

## Motivation and Context
- Improve performance.
- Following nanoframework/System.Math#151.

## How Has This Been Tested?
- On device
- Unit tests

## Types of changes

- [X] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [X] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [X] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [X] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

File: nanoFramework.CoreLibrary.NoReflection/CoreLibrary.NoReflection.nfproj [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="Current" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="..\packages\Nerdbank.GitVersioning.3.5.119\build\Nerdbank.GitVersioning.props" Condition="Exists('..\packages\Nerdbank.GitVersioning.3.5.119\build\Nerdbank.GitVersioning.props')" />
+  <Import Project="..\packages\Nerdbank.GitVersioning.3.6.133\build\Nerdbank.GitVersioning.props" Condition="Exists('..\packages\Nerdbank.GitVersioning.3.6.133\build\Nerdbank.GitVersioning.props')" />
```
Comment (josesimoes) [inline nanoFramework.CoreLibrary.NoReflection/CoreLibrary.NoReflection.nfproj:3]: PRs should be atomic and contain only the changes they are meant to have.
Please extract these reference update and go for a separate PR.

Comment (CoryCharlton) [reply nanoFramework.CoreLibrary.NoReflection/CoreLibrary.NoReflection.nfproj]: Done 👍

**Review by josesimoes — APPROVED** [review]
LGTM!
